### PR TITLE
Patch 159 (fail fast)

### PR DESCRIPTION
**Pull request by devinmarieb**
# Description
Updates to Sebastian's branch for preview.

# Reasons
First edit on the first draft.

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

**Comment by devinmarieb** [comment]
@sebastian-lerner any new changes, please use this PR to edit the commits. I couldn't directly pull down your branch and keep using the same PR you had been working on, but if you look above you can still see your commit history here. I have two basic questions so far:

- In the FAQ section we only call out one orb, which is a little weird. If we have only tested one orb internally, I think we need to add a note that other orbs won't work, or we can't guarantee that other orbs will work.
- Do you already have the diagrams ready to be added in? If so I can add them in now. If not we can do it later.

**Comment by sebastian-lerner** [comment]
@devinmarieb got it thank you!  we can add a note that other orbs haven't been tested. the reason i call out this orb is because it's a super common orb for running tests on circleci so i was trying to anticipate questions that were likely to come up.

diagrams are ready but let me know if you think they are good enough.  attaching them here

fail_fast_without_splitting
fail_fast_with_splitting

**Comment by sebastian-lerner** [comment]
the images have different style...sorry...the first one i didn't create and i dont have the source.  and i'm not great at creating diagrams... should i try to refine them?

**Comment by devinmarieb** [comment]
@sebastian-lerner These are fine for a preview. I think ideally they are refined for when the feature moves to GA

**Comment by sebastian-lerner** [comment]
we may want to add a note that "plugin manager" is a circleci component that manages state between batches. i wish i could change the name on that box..."plugin manager" is confusing

File: jekyll/_cci2/fail-fast.adoc [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+:page-description: A new way to tighten your feedback loop when running tests on CircleCI.
+:icons: font
+:toc: macro
+:toc-title:
+
+
+WARNING: This feature is in **Open preview**, use at your own risk. This feature is not guaranteed to move to general availability. For questions and/or issues, please comment on our link:[Community forum] (add link).
+
+[#motivation-and-introduction]
+== Motivation and Introduction
```
Comment (kbatuigas) [inline jekyll/_cci2/fail-fast.adoc:18]:
```suggestion
== Motivation and introduction
```

File: jekyll/_cci2/fail-fast.adoc [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+As a developer, speeding up your CI feedback loop is an excellent way to stay in your flow. For jobs with lengthy running tests, where even one failed test requires immediate attention, waiting for feedback until all tests have run to completion results in wasted time and unoptimized credit usage.
+
+Now in open preview is a new functionality to fail tests faster. You can configure your testing jobs to stop executing shortly after finding the first test failure.
+
+Unlike fail-fast functionality built into specific test runners, the configuration settings described below are compatible out-of-the-box with CircleCI's xref:test-splitting-tutorial#[intelligent test splitting] and xref:parallelism-faster-jobs#[parallelism]. Previously, it was non-trivial for a user to instruct CircleCI to terminate a parallel run (xref:runner-faqs#what-is-a-CircleCI-task-vs-a-job[task]) because a separate parallel run (task) found a test failure. The functionality described below is generic to any test runner.
+
+[#use-cases]
+== Use Cases
```
Comment (kbatuigas) [inline jekyll/_cci2/fail-fast.adoc:27]:
```suggestion
== Use cases
```

File: jekyll/_cci2/fail-fast.adoc [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+[#verify-the-configuration]
+==== Verify the configuration
+
+If the `--verbose` setting is enabled, you should see in your step output a description of the number of batches processed. This is an indication that the job has been configured successfully with the fail-fast functionality.
+
+[#batching]
+== Batching
+
+Batches are groups of tests that have been divided, and will report status after completion. If a test failure is found when executing the group of tests in a given batch, the batch will return a failed status to CircleCI. With fail-fast is enabled, once a batch has returned a failed status, CircleCI will prevent any subsequent batches from kicking off. Batch counts are set to `1` by default.
```
Comment (kbatuigas) [inline jekyll/_cci2/fail-fast.adoc:106]: "**With fail-fast enabled**"? Or "**When fail-fast is enabled**"?

File: jekyll/_cci2/fail-fast.adoc [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+If no test splitting is enabled, batches execute sequentially as shown in the diagram shows below. ("Plugin Manager" is a CircleCI component that manages state between batches).
```
Comment (kbatuigas) [inline jekyll/_cci2/fail-fast.adoc:108]:
```suggestion
If no test splitting is enabled, batches execute sequentially as shown in the diagram below. ("Plugin Manager" is a CircleCI component that manages state between batches.)
```

File: jekyll/_cci2/fail-fast.adoc [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+:toc: macro
+:toc-title:
+
+
+WARNING: This feature is in **Open preview**, use at your own risk. This feature is not guaranteed to move to general availability. For questions and/or issues, please comment on our link:[Community forum] (add link).
+
+[#motivation-and-introduction]
+== Motivation and Introduction
+
+As a developer, speeding up your CI feedback loop is an excellent way to stay in your flow. For jobs with lengthy running tests, where even one failed test requires immediate attention, waiting for feedback until all tests have run to completion results in wasted time and unoptimized credit usage.
```
Comment (kbatuigas) [inline jekyll/_cci2/fail-fast.adoc:20]:
```suggestion
For jobs with lengthy running tests, where even one failed test requires immediate attention, waiting for feedback until all tests have run to completion results in wasted time and unoptimized credit usage.
```
I think the first sentence can be dropped for conciseness.

File: jekyll/_cci2/fail-fast.adoc [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+
+Now in open preview is a new functionality to fail tests faster. You can configure your testing jobs to stop executing shortly after finding the first test failure.
```
Comment (kbatuigas) [inline jekyll/_cci2/fail-fast.adoc:22]: I am leaning towards rephrasing the first sentence. Something along the lines of... "CircleCI is releasing in open preview a new functionality to fail tests faster." I think the "now" makes it sound more like marketing copy.

**Review by kbatuigas — CHANGES_REQUESTED** [review]

**Review by kbatuigas — APPROVED** [review]
🙌

**Pull request merged**